omod/src/main/java/org/openmrs/module/owa/web/controller/OwaRestController.java

```diff
@@ -151,77 +157,110 @@
 
 	@RequestMapping(value = "/rest/owa/installapp", method = RequestMethod.POST)
 	@ResponseBody
-	public List<App> install(@RequestBody InstallAppRequestObject urlObject, HttpServletRequest request, HttpServletResponse response) throws IOException {
+	public ResponseEntity<List<App>> install(@RequestBody InstallAppRequestObject urlObject,
+	        HttpServletRequest request, HttpServletResponse response) throws IOException {
+		String message;
+		String fileName = null;
 		List<App> appList = new ArrayList<>();
+		Boolean checkInstall = false;
 
-		String url = urlObject.getUrlValue();
-		String fileName = urlObject.getFileName();
-		if(fileName == null || fileName == ""){
-			throw new  NullPointerException(
-				"File name must be passed for installation"
-			);
-		}
-		URL downloadUrl = null;
-		if (ResourceUtils.isUrl(url)) {
-			downloadUrl = new URL(url);
-		}
 		if (Context.hasPrivilege("Manage OWA")) {
-			String message;
 			HttpSession session = request.getSession();
-			if (!url.isEmpty()) {
+			URL downloadUrl = null;
+			String installUrl = urlObject.getUrlValue();
+
+			if (!installUrl.contains(".zip")) {
+				message = messageSourceService.getMessage("Invalid URL");
+				log.warn("Invalid URL to OWA download");
+				session.setAttribute(WebConstants.OPENMRS_ERROR_ATTR, message);
+				response.sendError(400, message);
+				return new ResponseEntity<List<App>>(HttpStatus.BAD_REQUEST);
+			}
+
+			try {
+				downloadUrl = ResourceUtils.getURL(installUrl);
+				fileName = getPassedFileName(installUrl);
+				if (fileName == null || fileName.isEmpty()) {
+					message = messageSourceService.getMessage("Installation URL doesnot specify download file");
```

Comment:
Do we have a message with that code or key? https://github.com/openmrs/openmrs-core/blob/master/api/src/main/resources/messages.properties

Reply:
Same followup comment as for the above.

Reply:
Have added the terms to `message.properties` in the OWA module. I have also updated the messages to be more descriptive